Tests/UnitTests/Sources/GREYActionsTest.m

```diff
@@ -198,5 +198,35 @@ - (void)testInvalidTapActionSucceedsAfterDisablingConstraints {
   }
 }
 
+- (void)testReplaceTextSucceedsOnUITextField {
+    NSString *textToReplace = @"A String";
+    NSError *error;
+
+    UITextField *textfield = [[UITextField alloc] init];
+
+    id<GREYAction> replace = [GREYActions actionForReplaceText:textToReplace];
+    [replace perform:textfield error:&error];
+
+    XCTAssertNil(error);
+
+    XCTAssertEqualObjects(textToReplace, textfield.text);
+}
+
+- (void)testReplaceTextFailsOnUIView {
+    NSString *textToReplace = @"A String";
+    NSError *error;
+
+    UIView *label = [[UIView alloc] init];
+
+    id<GREYAction> replace = [GREYActions actionForReplaceText:textToReplace];
+    [replace perform:label error:&error];
+
+    NSString *reasonSubstring = @"Action could not be performed on element";
```

Comment:
I updated the constraint which improves the error when someone tries to replace text on an element that doesn't define a setText method.

Actual failure handling requires the ability to verify the text was set correctly. EarlGrey doesn't have an API to get the text on both web/native elements. We also don't do that for the typeText action.